R/geom-defaults.r

```diff
@@ -29,27 +31,63 @@
 #'   geom_function(fun = dnorm, color = "red")
 #'
 #' # reset default
-#' update_stat_defaults("bin", aes(y = after_stat(count)))
-#'
+#' update_stat_defaults("bin", NULL)
+
 #' @rdname update_defaults
 update_geom_defaults <- function(geom, new) {
-  g <- check_subclass(geom, "Geom", env = parent.frame())
-  old <- g$default_aes
-  new <- rename_aes(new)
-  new_names_order <- unique(c(names(old), names(new)))
-  new <- defaults(new, old)[new_names_order]
-  g$default_aes[names(new)] <- new
-  invisible()
+  if (is.null(new)) {
+
+    vec_assert(geom, character(), 1)
```

Comment:
Shouldn't you use the check infrastructure here?

Reply:
Good point, also forgot the case where `geom = GeomPoint` as ggproto case. Fixed now.